knowledge_base/new-index-replica-before-upgrade-consistent-documents.json

```diff
@@ -0,0 +1,4 @@
+{
+  "id": "new-index-replica-before-upgrade-consistent-documents",
+  "description": "An index and its replica should contain the same documents when the replica is created before a migration or upgrade"
```

Comment:
I would probably agree with this. As I was writing these expectations it was often the thought that I could easily change a small piece of a given expectation and probably have another valuable expectation to add as well so the level of detail can be desirable. 

I think the tricky part of this is how much detail is appropriate for the expectation model, as I accommodate more and more details the sentence structure begins to be not as pleasant and having a standard DSL starts to look a lot like the robot tests themselves and/or creates a little larger gap for new committers to add to.

I'd very interested to see an example of your thoughts @gregschohn for an informal DSL which sounds like it could potentially be a nice middle ground? I had played with doing a sample model that I for whatever reason gravitated toward SQL-like but didn't feel much better than the robot tests themselves to understand
```
GIVEN
CLUSTER source_cluster
CLUSTER target_cluster
INDEX test-index ON source_cluster 
	WITH DATA well-formed random number of documents n 
	WITH SETTING zero replicas

WHEN
test-index CHANGES SETTING one replicas
source_cluster UPGRADES|MIGRATES to target_cluster

THEN
Each DOCUMENT IN INDEX test-index ON target_cluster ON node with primary MATCHES
Each DOCUMENT IN INDEX test-index ON target_cluster ON node with replica
```